configure.ac

```diff
@@ -821,33 +821,41 @@ AC_SUBST(LEMON)
 AM_CONDITIONAL([WITH_LEMON], [test "$lemon_available" = "yes"])
 
 # libedit
-AC_ARG_ENABLE(libedit,
-  [AS_HELP_STRING([--disable-libedit],
-    [use libedit for console. [default=auto-detect]])],
-  [enable_libedit="$enableval"],
-  [enable_libedit="auto"])
-if test "x$enable_libedit" != "xno"; then
-  m4_ifdef([PKG_CHECK_MODULES], [
-    PKG_CHECK_MODULES([LIBEDIT], [libedit >= 3.0],
-      [LIBS_SAVE="$LIBS"
-       LDFLAGS_SAVE="$LDFLAGS"
-       LDFLAGS="$LIBEDIT_LIBS $LDFLAGS"
-       AC_SEARCH_LIBS(el_wline, edit,
-                      [libedit_available=yes],
-                      [libedit_available=no])
-       LIBS="$LIBS_SAVE"
-       LDFLAGS="$LDFLAGS_SAVE"],
-      [libedit_available=no])
-    ],
-    [libedit_available=no])
```

Comment:
Could you remain pkg-config support?
Could you use pkg-config first then use `AC_CHECK_LIB()` fallback? Because pkg-config is more generic than `AC_CHECK_LIB()`.